worldedit-core/src/main/java/com/sk89q/worldedit/function/util/ChangeLimitingFilter.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * WorldEdit, a Minecraft world manipulation toolkit
+ * Copyright (C) sk89q <http://www.sk89q.com>
+ * Copyright (C) WorldEdit team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldedit.function.util;
+
+import com.sk89q.worldedit.MaxChangedBlocksException;
+import com.sk89q.worldedit.WorldEditException;
+import com.sk89q.worldedit.function.RegionFunction;
+import com.sk89q.worldedit.math.BlockVector3;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A filter that limits the number of changes that can be made.
+ */
+public class ChangeLimitingFilter implements RegionFunction {
+
+    private final RegionFunction function;
+    private final int limit;
+    private int count = 0;
+
+    /**
+     * Create a new change limiting filter.
+     *
+     * @param limit the limit
+     * @param function the function
+     */
+    public ChangeLimitingFilter(int limit, RegionFunction function) {
+        checkNotNull(function);
+        checkArgument(limit >= -1, "limit >= -1 required");
```

Comment:
Would it make more sense to enforce `>= 0` instead, and you should just not add this if your limit is less than that? It would reduce potential branches in a hot area.

Reply:
So I've not changed this, but I've made our usage of it be conditionally applied. IMO it doesn't make sense to restrict this beyond valid values

Reply:
I don't really get why a negative value is valid though. If it rejected it, the branch in apply could be removed.